pilot/pkg/networking/core/cluster_traffic_policy.go

```diff
@@ -275,14 +272,20 @@ func applyLoadBalancer(c *cluster.Cluster, lb *networking.LoadBalancerSettings,
 	ApplyRingHashLoadBalancer(c, lb)
 }
 
-func applyLocalityLoadBalancer(locality *core.Locality, proxyLabels map[string]string, cluster *cluster.Cluster,
+func applyLocalityLoadBalancer(locality *core.Locality, proxyLabels map[string]string, c *cluster.Cluster,
 	localityLB *networking.LocalityLoadBalancerSetting,
 ) {
 	// Failover should only be applied with outlier detection, or traffic will never failover.
-	enableFailover := cluster.OutlierDetection != nil
-	if cluster.LoadAssignment != nil {
+	enableFailover := c.OutlierDetection != nil
+	// set locality weighted lb config when locality lb is enabled, otherwise it will influence the result of LBPolicy like `least request`
+	if enableFailover || localityLB.GetDistribute() != nil {
+		c.CommonLbConfig.LocalityConfigSpecifier = &cluster.Cluster_CommonLbConfig_LocalityWeightedLbConfig_{
```

Comment:
I am a bit confused. This code only applies to CDS, not EDS. Why is it different? I would expect this logic to be in the common `ApplyLocalityLoadBalancer`?

But weirdly it seems to behave correctly even for EDS, which basically means we never set this but it works..?

Reply:
Not sure i understand. This function applies to each cluster(outbound only), so even eds typed cluster, it will be applied

Reply:
Ohh sorry.. I was confused. thanks.